From d70abdeb761df736c99583045cdd754d84d85159 Mon Sep 17 00:00:00 2001
From: "Gross, Mark" <mark.gross@intel.com>
Date: Thu, 4 Jan 2018 11:09:51 -0800
Subject: [PATCH 408/548] REVERTME disable user mode non-root access to the
 eBPF JIT

This change is to block the "meltdown" side channel exploit instance using
the eBPF JIT as an attack vector.

We need to revert this when a proper fix to the side channel attacks are made
available.

Signed-off-by: Mark Gross <mark.gross@intel.com>
Signed-off-by: Saul Wold <sgw@linux.intel.com>
---
 kernel/bpf/syscall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/bpf/syscall.c b/kernel/bpf/syscall.c
index 25d074920a00..7bfeaca34139 100644
--- a/kernel/bpf/syscall.c
+++ b/kernel/bpf/syscall.c
@@ -1455,7 +1455,7 @@ SYSCALL_DEFINE3(bpf, int, cmd, union bpf_attr __user *, uattr, unsigned int, siz
 	union bpf_attr attr = {};
 	int err;
 
-	if (!capable(CAP_SYS_ADMIN) && sysctl_unprivileged_bpf_disabled)
+	if (!capable(CAP_SYS_ADMIN))
 		return -EPERM;
 
 	err = check_uarg_tail_zero(uattr, sizeof(attr), size);
-- 
2.16.1

